### ISSUE TITLE
doxygen 1.8.15 segfault

### ISSUE THREAD

**Issue by pgajdos** [doxygen/doxygen#6749]
I get a segfault during ldb (https://ldb.samba.org/)  build.

Testcase: [testcase.zip](https://github.com/doxygen/doxygen/files/2733280/testcase.zip)

It is not quite minimal, perhaps I could try to create smaller one. 

```
$ gdb --args doxygen Doxyfile
[..]
/testcase/ldb.h:2337: warning: Member ldb_val_string_cmp(const struct ldb_val *v, const char *str) (function) of file ldb.h is not documented.

Program received signal SIGSEGV, Segmentation fault.
0x000055555589d1ca in latexFilterURL (s=<optimized out>) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/util.cpp:7045
7045	  while ((c=*p++))
(gdb) bt
#0  0x000055555589d1ca in latexFilterURL (s=<optimized out>) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/util.cpp:7045
#1  0x0000555555a99d5d in LatexDocVisitor::visitPre (this=0x555556784d00, href=0x555556782270) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/qtools/qcstring.h:592
#2  0x0000555555a6dbe0 in CompAccept<DocHRef>::accept (this=0x555556782270, v=0x555556784d00) at /usr/include/bits/stdio2.h:104
#3  0x0000555555a681f1 in CompAccept<DocPara>::accept (this=<optimized out>, v=0x555556784d00) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/qtools/qlist.h:136
#4  0x0000555555a3dff1 in DocSimpleSect::accept (this=0x555556784a00, v=0x555556784d00) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/docparser.cpp:4543
#5  0x0000555555a681f1 in CompAccept<DocPara>::accept (this=<optimized out>, v=0x555556784d00) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/qtools/qlist.h:136
#6  0x00005555559df111 in CompAccept<DocRoot>::accept (this=<optimized out>, v=0x555556784d00) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/qtools/qlist.h:136
#7  0x00005555557bc0ed in LatexGenerator::writeDoc (this=0x5555566a81e0, n=0x555556782b60, ctx=<optimized out>) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/latexgen.cpp:2182
#8  0x00005555558348b0 in OutputList::writeDoc (this=<optimized out>, root=0x555556782b60, ctx=0x5555565699b0, md=0x5555565ce740)
    at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/outputlist.cpp:170
#9  0x0000555555834a1b in OutputList::generateDoc (this=this@entry=0x555556563260, fileName=0x5555567480b8 "/testcase/ldb.h", startLine=startLine@entry=724, 
    ctx=ctx@entry=0x5555565699b0, md=md@entry=0x5555565ce740, docStr=..., indexWords=true, isExample=false, exampleName=0x0, singleLine=false, linkFromIndex=false)
    at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/outputlist.cpp:153
#10 0x000055555580b95d in MemberDef::writeDocumentation (this=this@entry=0x5555565ce740, ml=ml@entry=0x555556599560, memCount=<optimized out>, memTotal=memTotal@entry=1, ol=..., 
    scName=scName@entry=0x7fffffffe422 "ldb.h", container=<optimized out>, inGroup=false, showEnumValues=false, showInline=false)
    at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/qtools/qcstring.h:592
#11 0x000055555581b9b1 in MemberList::writeDocumentation (this=this@entry=0x555556599560, ol=..., scopeName=0x7fffffffe422 "ldb.h", container=container@entry=0x5555565699b0, 
    title=title@entry=0x55555655fd18 "Macro Definition Documentation", showEnumValues=showEnumValues@entry=false, showInline=false)
    at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/memberlist.cpp:741
#12 0x00005555556ad322 in FileDef::writeMemberDocumentation (this=0x5555565699b0, ol=..., lt=<optimized out>, title=...)
    at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/qtools/qcstring.h:592
#13 0x00005555556ae088 in FileDef::writeDocumentation (this=0x5555565699b0, ol=...) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/filedef.cpp:895
#14 0x000055555568e93e in generateOutput() () at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/doxygen.cpp:8087
#15 0x0000555555646c7e in main (argc=2, argv=0x7fffffffe848) at /usr/src/debug/doxygen-1.8.15-142.1.x86_64/src/main.cpp:38
(gdb)
```

I do not get such segfault for 1.8.14 with this testcase.

Let me know how can I help.

**Comment by albert-github** [comment]
I see that the problem is in the LaTeX generation part (and the crash shouldn't happen and should be fixed).
In the Doxyfile I see that you explicitly enable HTML and man output, but don't mention the LaTeX output there, is this on purpose ? (when you don't need the LaTeX a workaround here is to set GENERATE_LATEX to NO).

**Comment by albert-github** [comment]
Looks like the offending line in the input is line 727:
```
\sa <a href="">Microsoft documentation of this OID</a>
```

An empty href...

**Comment by pgajdos** [comment]
Yep, I came to smaller testcase:
```
/**
   \file ldb.h Samba's ldb database
*/
/**
   \sa <a href="">Microsoft documentation of this OID</a>
*/
#define LDB_CONTROL_RODC_DCPROMO_NAME   "rodc_join"

```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6751

**Comment by pgajdos** [comment]
The change fixes the segfault for me. Thanks a lot for the quick response!

**Comment by albert-github** [comment]
Code has been integrated in master on github.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
